rides/__init__.py

```diff
@@ -381,9 +395,76 @@ def leave_ride(car_id, rider_username):
     username = current_user.id
     car = Car.query.get(car_id)
     rider = Rider.query.filter(Rider.username == rider_username, Rider.car_id == car_id).first()
+    if rider.username == username and rider is not None:
+        db.session.delete(rider)
+        car.current_capacity -= 1
+        notify_opening( car.event_id, car.name, car_id )
+        db.session.add(car)
+        db.session.commit()
+    return redirect(url_for('index'))
+
+# Automatically leave a ride and join different ride, used for Needs a Ride Notifications
+@app.route('/autojoin/<string:leave_id>/<string:join_id>/<string:user>', methods=['GET'])
+@login_required
+def autojoin(leave_id, join_id, user):
+    # LEAVE
+    username = current_user.id
+    car = Car.query.get(leave_id)
+    rider = Rider.query.filter(Rider.username == user, Rider.car_id == leave_id).first()
     if rider.username == username and rider is not None:
         db.session.delete(rider)
         car.current_capacity -= 1
         db.session.add(car)
         db.session.commit()
+    # JOIN
+    #username = current_user.id
+    name = current_user.firstname + " " + current_user.lastname
+    slack = current_user.slack
+    email = current_user.email
+    acc_type = current_user.acc_type
+    car = Car.query.get(join_id)
+    event = Event.query.get(car.event_id)
+    attempted_username = user
+    if attempted_username == username:
+        for c in event.cars:
+            incar = ( c.username == username ) or ( username in c.riders )
+        if (car.current_capacity < car.max_capacity or car.max_capacity == 0) and not incar:
+            rider = Rider( username, name, join_id, slack, email, acc_type )
+            car.current_capacity += 1
+            db.session.add(rider)
+            db.session.add(car)
+            db.session.commit()
     return redirect(url_for('index'))
+
+# Notify user of opening in event, if slack fails go to email
+def notify_opening(event_id, driver_name, car_id):
+    event = Event.query.get(event_id)
+    event_name = event.name
+    need_ride_car = Car.query.filter( Car.event_id == event_id and Car.username == '∞' ).first()
+    need_ride = Rider.query.filter( Rider.car_id == need_ride_car.id ).all()
+    for rider in need_ride:
+        if rider is not None:
+            name = rider.name
+            url = "https://rideboard.csh.rit.edu/autojoin/"
+            url += str(need_ride_car.id) + "/" + str(car_id)  + "/" + rider.username
+            try:
+                client.chat_postMessage(channel=rider.slack,text="Hello " + name +
+                    ",\n\nThere is a ride available for " + event_name +
+                    "!\nDriver of the available car is " + driver_name +
+                    ".\nGo to " + url + " to claim your spot!")
```

Comment:
Nit: I'd prefer the punctuation and the `\n`s to go at the end of the line of code as well. It would help to visually separate them in code.